src/sdf/small_vector.h

```diff
@@ -0,0 +1,397 @@
+// Copyright 2024 The Manifold Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// rewrite of https://github.com/p-ranav/small_vector
```

Comment:
Originally I copied it from the repository mentioned, but it turns out it was too buggy and misses some required functions, so I changed it quite a bit.